pkgs/applications/misc/hello/default.nix

```diff
@@ -1,16 +1,17 @@
-{ lib
+{ callPackage
+, lib
 , stdenv
 , fetchurl
 , testVersion
 , hello
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "hello";
   version = "2.10";
 
   src = fetchurl {
-    url = "mirror://gnu/hello/${pname}-${version}.tar.gz";
+    url = "mirror://gnu/hello/${finalAttrs.pname}-${finalAttrs.version}.tar.gz";
```

Comment:
```suggestion
    url = "mirror://gnu/hello/${finalAttrs.public.name}.tar.gz";
```

Reply:
I disagree with this suggested change. If I'm overriding the package I should have some means of giving it a different name (so I can visually distinguish it) without breaking the `src` attribute. With the code as written, I can provide a `name` attr and leave `pname` alone. This is admittedly slightly odd, but it would work. Alternatively the package could just hardcode the name `hello` in the URL, like it does in the directory name, and that would leave both `name` and `pname` free for use.

Reply:
I've hardcoded the use of `pname` in the url, because it doesn't make sense to DRY things we don't control and are just a given.